From 65ade99fb2b858811ed0b0447b79b821c09a8037 Mon Sep 17 00:00:00 2001
From: Ildar Mulyukov <ildar.mulyukov@bell-sw.com>
Date: Sat, 21 Jun 2025 15:26:24 +0500
Subject: [PATCH] runtime-id: fix nonPortableRid according to dotnet's way

align with https://github.com/dotnet/dotnet/blob/v8.0.17/eng/common/native/init-distro-rid.sh#L28
This fixes the Rid for rolling release distros that have non-numeric
VERSION_ID in /etc/os-release.
---
 runtime-id | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/runtime-id b/runtime-id
index 7aff7da..1aa2bc0 100755
--- a/runtime-id
+++ b/runtime-id
@@ -82,5 +82,13 @@ else
             ;;
     esac
 
-    echo "${ID}.${rid_version}-${arch}"
+    if [[ "${rid_version:-}" =~ ^([[:digit:]]|\.)+$ ]]; then
+        nonPortableRid="${ID}.${rid_version}-${arch}"
+    else
+        # Rolling release distros either do not set VERSION_ID, set it as blank or
+        # set it to non-version looking string (such as TEMPLATE_VERSION_ID on ArchLinux);
+        # so omit it here to be consistent with everything else.
+        nonPortableRid="${ID}-${arch}"
+    fi
+    echo "${nonPortableRid}"
 fi
-- 
2.42.4

